src/ansys/fluent/core/launcher/watchdog.py

```diff
@@ -0,0 +1,302 @@
+
+    kwargs = {"env": watchdog_env, "stdin": subprocess.DEVNULL, "close_fds": True}
+
+    if os.name == "nt":
+        kwargs.update(shell=True)
+        # https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
+        # https://docs.python.org/3/library/subprocess.html#windows-constants
+        flags = 0
+        flags |= subprocess.CREATE_NO_WINDOW
+        flags |= subprocess.CREATE_NEW_PROCESS_GROUP
+        flags |= subprocess.DETACHED_PROCESS
+        flags |= subprocess.CREATE_BREAKAWAY_FROM_JOB
+        flags |= subprocess.SW_HIDE
+        kwargs.update(creationflags=flags)
+
+    if os.name == "posix":
+        kwargs.update(start_new_session=True)
+
+    if env_watchdog_debug in [1, "1", "ON"] and os.name != "nt":
+        kwargs.update(
+            stdout=open(f"pyfluent_watchdog_out_{watchdog_id}.log", mode="w"),
+            stderr=open(f"pyfluent_watchdog_err_{watchdog_id}.log", mode="w"),
+            bufsize=1,
+        )
+    else:
+        kwargs.update(stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+    logger.debug(f"kwargs: {kwargs}")
+
+    if os.name == "nt":
+        # https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start
+        os_cmd = ["start"]
+    else:
+        os_cmd = []
+
+    cmd_send = os_cmd + command_list
+    logger.debug(f"Command list: {cmd_send}")
+
+    init_file = Path(WATCHDOG_INIT_FILE.format(watchdog_id))
+    if init_file.is_file():
+        init_file.unlink()
+
+    subprocess.Popen(cmd_send, **kwargs)
+
+    logger.debug(f"Waiting for Watchdog to initialize, then proceeding...")
+    i = 0.0
+    success = False
+    while i <= 10:
+        if init_file.is_file():
+            success = True
+            init_file.unlink()
+            break
+        i += 0.5
+        time.sleep(0.5)
+    if not success:
+        logger.warning(
+            "PyFluent Watchdog did not initialize correctly, proceeding without it..."
+        )
+
+
+if __name__ == "__main__":
+    print(
+        "Starting PyFluent Watchdog process, do not manually close or terminate this process, "
+        "it will automatically exit once finished.".upper()
+    )
+
+    watchdog_id = sys.argv[5]
+
+    import psutil
+
+    launcher_pid = int(sys.argv[1])
+
+    from ansys.fluent.core.fluent_connection import FluentConnection, get_container_ids
+
+    # Configure logger for Watchdog process
+    log_config = pyfluent.logging.get_default_config()
+    log_config["handlers"]["pyfluent_file"][
+        "filename"
+    ] = f"pyfluent_watchdog_{watchdog_id}.log"
+
+    logger = pyfluent.logging.get_logger("pyfluent.watchdog")
+
+    if os.getenv("PYFLUENT_WATCHDOG_DEBUG", "off").upper() in [1, "1", "ON"]:
+        pyfluent.logging.enable(custom_config=log_config)
+        logger.setLevel("DEBUG")
+        logger.handlers = pyfluent.logging.get_logger(
+            "pyfluent.general"
+        ).handlers  # using same handlers as already defined
```

Comment:
This is an example of how the current logging setup can be customized when e.g. running scripts (in this case being used for Watchdog as it runs outside and separate from main PyFluent)